Program.cs

```diff
@@ -1,8 +1,12 @@
+using SendGrid.Extensions.DependencyInjection;
+
 var builder = WebApplication.CreateBuilder(args);
 
 // Add services to the container.
 builder.Services.AddRazorPages();
-
+builder.Services.AddSendGrid(options =>
+    options.ApiKey = builder.Configuration["SendGridApiKey"]
```

Comment:
Use `SendGridApiKey` not `SendGridAPIKey`.
Acronyms like API should follow the same camelCasing or PascalCasing rules, and not be all caps.